### ISSUE TITLE
Longevity 1000 keyspaces - request throughput fluctuations during nodetool repair (with hinted handoff disabled)

### ISSUE THREAD

**Issue by piodul** [scylladb/scylla#5647]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): 3.2.0
Cluster size: 5
OS (RHEL/CentOS/Ubuntu/AWS AMI): 

On a short Longevity 1000 keyspaces test run, with only TerminateAndReplaceNode nemesis enabled, inconsistent throughput could be observed. The nemesis ran twice, and in both cases request throughput was fluctuating:

![72779851-eca64e80-3c25-11ea-8551-f3915d4c631a](https://user-images.githubusercontent.com/4950936/73098911-a7ff0980-3eea-11ea-88a0-f64e1ce913df.png)

Please note that this is a different case than #4981, where throughput decrease is caused by hinted handoff. In this test run, hinted handoff was explicitly _disabled_.

Monitoring data:

http://34.241.207.103:3000/d/q4I08GsWk/scylla-per-server-metrics-nemesis-3-2?orgId=1&from=1579556598406&to=1579588910028&var-by=instance&var-cluster=&var-dc=All&var-node=All&var-shard=All&var-sct_tags=DisruptionEvent

Test run artifacts:
```
+--------------------------------------------------------------------------------------------------------------------------------------------+
|  Collected logs by test-id: 8972866b-bd94-4ca6-a173-ca66379823c4 Directory: /home/fruch/sct-results/20200121-100847-130691/collected_logs  |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| Cluster set | Link                                                                                                                         |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/8972866b-bd94-4ca6-a173-ca66379823c4/20200121_102756/db-cluster-8972866b.zip  |
| monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/8972866b-bd94-4ca6-a173-ca66379823c4/20200121_102756/monitor-set-8972866b.zip |
| loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/8972866b-bd94-4ca6-a173-ca66379823c4/20200121_102756/loader-set-8972866b.zip  |
| sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/8972866b-bd94-4ca6-a173-ca66379823c4/20200121_102756/sct-runner-8972866b.zip  |
+-------------+------------------------------------------------------------------------------------------------------------------------------+

to get monitor data: 
hydra investigate show-monitor 8972866b-bd94-4ca6-a173-ca66379823c4
```

**Comment by slivne** [comment]
5 nodes is not large :)

**Comment by slivne** [comment]
@asias I hnk we have better row level repair bandwidth control in 3.3 / 4.0 - is that so ?

@roydahan when we get to this tests  please post an update on 4.0 status

**Comment by asias** [comment]
> @asias I hnk we have better row level repair bandwidth control in 3.3 / 4.0 - is that so ?

We still do not have network scheduling. We can not control the network bandwidth usage between cql and repair traffic. 

> @roydahan when we get to this tests please post an update on 4.0 status

**Comment by roydahan** [comment]
@fgelcer now that you've run the test on 4.0, can you please report back?

**Comment by fgelcer** [comment]
> @fgelcer now that you've run the test on 4.0, can you please report back?

the test run terminate_and_replace_node nemesis, but it miserably failed with #6248 , so at the end, we finished the nemesis with one node less than we started (started with 6 nodes).
In addition to it, our monitor node ran out of memory, and that was addressed by https://github.com/scylladb/scylla-cluster-tests/pull/2099

**Comment by glommer** [comment]
Hi QA friends: are you going to re-do this?
We'll soon start a sprint to try and solve all issues related to effects of repair and maintenance in the cluster.

If this is still happening with row level repair we'll prioritize. Let us know

**Comment by roydahan** [comment]
@fgelcer please report results from the second run you had.
another option is to run again with TerminateAndReplaceNode nemesis.

**Comment by fgelcer** [comment]
last multi-keyspaces longevity job didn't run decommission nor terminate_and_replace nemesis.
i will re-run this job with terminate_and_replace nemesis only to gather more info about this.

**Comment by glommer** [comment]
Good luck @fgelcer ! run it from a 2m distance so you don't get covid-19

**Comment by fgelcer** [comment]
we had 1 success on terminate_and_replace and 1 failure.
the success one took around 3.5 hours.
during the 1st replace nemesis we have a running repair, and then we have suppressed messages, so no info about repair finishing:
```
2020-05-21T15:19:41+00:00  longevity-1000-keyspaces-multi-ke-db-node-f4dab6ac-7 !INFO    | scylla: [shard 7] compaction - Compacted 2 sstables to [/var/lib/scylla/data/keyspace877/standard1-57e4f6909b5511eaa10d00000000000a/mc-577-big-Data.db:level=0, ]. 862660 bytes to 826948 (~95% of original) in 505ms = 1.56MB/s. ~3328 total partitions merged to 3188.
2020-05-21T15:26:11+00:00  longevity-1000-keyspaces-multi-ke-db-node-f4dab6ac-7 !INFO    | journal: Suppressed 26039 messages from /scylla.slice/scylla-server.slice
```

the 2nd one failed with spot termination

**Comment by fgelcer** [comment]
> Good luck @fgelcer ! run it from a 2m distance so you don't get covid-19

i'm running it from few thousands of km, as it is running on North Ireland :-)

**Comment by fgelcer** [comment]
@roydahan , shall i re-run the job, with on_demand to have the terminate_and_replace running more than once?

**Comment by roydahan** [comment]
Yes, and you didn't report on the throughput fluctuations this bug is talking about...

**Comment by fgelcer** [comment]
there is a different fluctuation in the latest test, but as it got spot termination, then we don't know what was supposed to happen:

![throughtput](https://user-images.githubusercontent.com/32832995/82759132-f1f1f080-9df3-11ea-9373-78fb06fc2b51.png)

i will re-run it with on_demand to try to reproduce the issue

**Comment by fgelcer** [comment]
the monitor is alive for a limited time, and can be accessed in this [address](http://3.250.196.71:3000/d/UWA5JVzGk/scylla-per-server-metrics-nemesis-master?orgId=1&refresh=30s&from=now-12h&to=now&var-by=instance&var-cluster=&var-dc=All&var-node=All&var-shard=All)

in any case, i took a [snapshot](https://snapshot.raintank.io/dashboard/snapshot/tewNOVYEIdrX3rrp7s6BJeAhem6KFZeq) of the monitor.

here is a command to create a local monitor with all information in the original one:
```
hydra investigate show-monitor --test-id 8da2a87a-0685-4f07-92f2-24185f186fb7
```

**Comment by fgelcer** [comment]
> the monitor is alive for a limited time, and can be accessed in this [address](http://3.250.196.71:3000/d/UWA5JVzGk/scylla-per-server-metrics-nemesis-master?orgId=1&refresh=30s&from=now-12h&to=now&var-by=instance&var-cluster=&var-dc=All&var-node=All&var-shard=All)
> 
> in any case, i took a [snapshot](https://snapshot.raintank.io/dashboard/snapshot/tewNOVYEIdrX3rrp7s6BJeAhem6KFZeq) of the monitor.
> 
> here is a command to create a local monitor with all information in the original one:
> 
> ```
> hydra investigate show-monitor --test-id 8da2a87a-0685-4f07-92f2-24185f186fb7
> ```

@glommer , @asias

**Comment by slivne** [comment]
this is the screenshot from the snapshot  - so we still have a clear drop 

![repair](https://user-images.githubusercontent.com/3465480/104850187-43f10480-58f6-11eb-8a26-498f3adee75b.png)

**Comment by denesb** [comment]
Closing as stale.
